plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/standard/StandardDeployCommandHandler.java

```diff
@@ -67,38 +64,8 @@ private boolean checkAppEngineRuntimeCompatibility(Shell shell, IProject project
         return false;
       }
 
-      DeployPreferences deployPreferences = getDeployPreferences(project);
-      if (AppEngineStandardFacet.usesObsoleteRuntime(facetedProject)
-          && !deployPreferences.getAllowObsoleteRuntime()) {
-        // Explicitly specify button IDs as MessageDialogWithToggle, unlike MessageDialog, assigns
-        // default IDs for custom buttons starting at 256
-        LinkedHashMap<String, Integer> buttonsWithIds = new LinkedHashMap<>();
-        buttonsWithIds.put(Messages.getString("deploy.button"), IDialogConstants.OK_ID);
-        buttonsWithIds.put(IDialogConstants.CANCEL_LABEL, IDialogConstants.CANCEL_ID);
-        String message = Messages.getString("obsolete.runtime.proceed", project.getName());
-        MessageDialogWithToggle dialog =
-            new MessageDialogWithToggle(
-                shell,
-                Messages.getString("obsolete.runtime.title"),
-                null,
-                message,
-                MessageDialog.WARNING,
-                buttonsWithIds,
-                0 /* index into buttonsWithIds of default button */,
-                Messages.getString("obsolete.runtime.toggle"),
-                false);
-        // For MessageDialog and MessageDialogWithToggle, open() returns the dialog's return code,
-        // and the return code is set to the ID of the selected button
-        if (dialog.open() == IDialogConstants.OK_ID && dialog.getToggleState()) {
-          try {
-            deployPreferences.setAllowObsoleteRuntime(true);
-            deployPreferences.save();
-          } catch (BackingStoreException ex) {
-            logger.log(
-                Level.WARNING, "Unable to check project use of obsolete App Engine runtime", ex);
-          }
-        }
-        return dialog.getReturnCode() == IDialogConstants.OK_ID;
+      if (AppEngineStandardFacet.usesObsoleteRuntime(facetedProject)) {
+        return false;
```

Comment:
I see one problem that when we identified an obsolete runtime, the deploy operation abruptly halts without any message to the user. (`checkProject()` in `execute()` simply returns null.) The user may believe that the deploy command isn't just working without knowing what the problem is. Somehow, we should let the user know.

Reply:
Good catch. So maybe I should put the dialog back here but change the message?

Reply:
I think that makes sense. The dialog style should change to have a single OK button.

Reply:
done